### ISSUE TITLE
Update outdated version of the Google Gson library dependency

### ISSUE THREAD

**Issue by sevokn** [cloudant/java-cloudant#292]
The java-cloudant project depends on the Google Gson library.

The version currently being used in java-cloudant, as can be seen in the Gradle configuration file https://github.com/cloudant/java-cloudant/blob/master/cloudant-client/build.gradle is 2.2.4, which is very old. It was released according to https://github.com/google/gson/blob/master/CHANGELOG.md in May 2013.

The most current version of the Gson library is 2.7, released in June 2016.

Is there any specific technical reason why the java-cloudant project uses a three-year-old outdated version of a (IMO) crucial dependency library? (After all, JSON is what Cloudant is all about.) The Gson changelog document lists several bug fixes and improvements in JSON handling since version 2.2.4 which all would surely benefit the users of the java-cloudant project.

(And sure, it's possible for a java-cloudant user to override outdated the Gson dependency in his own dependency settings of his own project that uses java-cloudant, but it would be more clear if java-cloudant would depend on a more current version of Gson out-of-the-box so that no separate configuration of Gson would be needed.)

**Comment by ricellis** [comment]
No one has reported to us any issues using the client with this particular version of the library so there hasn't been a need to upgrade it. Is there a specific bug fix in GSON that is impacting you?

**Comment by sevokn** [comment]
No, I don't have any specific issue in mind (at least yet) that would impact me directly and that would be fixed by a version upgrade of the Gson dependency. I just got started developing with the Cloudant Java API and while setting up the project I noticed the old version of the Gson dependency, and was just a bit surprised about it.

I just generally find it beneficial and meaningful to try to keep the dependency imports updated so that no technical debt is accumulated. A bug fix in a dependency is a free bug fix for your own application too.
